pkg/runner/runner.go

```diff
@@ -2,27 +2,61 @@ package runner
 
 import (
 	"errors"
+	"fmt"
 	"os"
 	"os/exec"
 	"strings"
 
+	"github.com/kelseyhightower/envconfig"
 	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
 	"github.com/kubeshop/testkube/pkg/executor/content"
 	"github.com/kubeshop/testkube/pkg/executor/output"
+	"github.com/kubeshop/testkube/pkg/executor/scrapper"
 )
 
+type Params struct {
+	Endpoint        string // RUNNER_ENDPOINT
+	AccessKeyID     string // RUNNER_ACCESSKEYID
+	SecretAccessKey string // RUNNER_SECRETACCESSKEY
+	Location        string // RUNNER_LOCATION
+	Token           string // RUNNER_TOKEN
+	Ssl             bool   // RUNNER_SSL
+}
+
 // NewRunner creates a new SoapUIRunner
 func NewRunner() *SoapUIRunner {
+	var params Params
+	err := envconfig.Process("runner", &params)
+	if err != nil {
+		panic(err.Error())
+	}
+
 	return &SoapUIRunner{
 		SoapUIExecPath: "/usr/local/SmartBear/EntryPoint.sh",
+		SoapUILogsPath: "/root/.soapuios/logs",
 		Fetcher:        content.NewFetcher(""),
+		Scrapper: scrapper.NewScrapper(
+			params.Endpoint,
+			params.AccessKeyID,
+			params.SecretAccessKey,
+			params.Location,
+			params.Token,
+			params.Ssl,
+		),
 	}
 }
 
 // SoapUIRunner runs SoapUI tests
 type SoapUIRunner struct {
 	SoapUIExecPath string
+	SoapUILogsPath string
 	Fetcher        content.ContentFetcher
+	Scrapper       Scrapper
+}
+
+// Scrapper is responsible for persisting the necessary artifacts
+type Scrapper interface {
```

Comment:
Why do we define it there instaed of scraper package?

Reply:
I will move it there. Do you think it would be worth to update the cypress executor to use the interface, and add it to the executor template as well?

Reply:
Think so, at the end, all executors will be moved in contrib api folder. This should simplify such changes

Reply:
I added it here: https://github.com/kubeshop/testkube/pull/1449
Not sure about the name though. Do you think it would be worth calling the interface Scraper and the implementation ContainerScraper?

Reply:
interface Scraper and implementation probably MinioScraper, because it's Minio implementation of Scraper interface

Reply:
updated, will do the cypress-executor as well